bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/handler/BoschIndegoHandler.java

```diff
@@ -36,46 +41,95 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import de.zazaz.iot.bosch.indego.DeviceCommand;
-import de.zazaz.iot.bosch.indego.DeviceStateInformation;
-import de.zazaz.iot.bosch.indego.DeviceStatus;
-import de.zazaz.iot.bosch.indego.IndegoAuthenticationException;
-import de.zazaz.iot.bosch.indego.IndegoController;
-import de.zazaz.iot.bosch.indego.IndegoException;
-
 /**
  * The {@link BoschIndegoHandler} is responsible for handling commands, which are
  * sent to one of the channels.
  *
  * @author Jonas Fleck - Initial contribution
+ * @author Jacob Laursen - Refactoring, bugfixing and removal of dependency towards abandoned library
  */
+@NonNullByDefault
 public class BoschIndegoHandler extends BaseThingHandler {
 
     private final Logger logger = LoggerFactory.getLogger(BoschIndegoHandler.class);
-    private final Queue<DeviceCommand> commandQueue = new LinkedList<>();
-
-    private ScheduledFuture<?> pollFuture;
+    private final HttpClient httpClient;
 
-    // If false the request is already scheduled.
-    private boolean shouldReschedule;
+    private @NonNullByDefault({}) IndegoController controller;
+    private @Nullable ScheduledFuture<?> pollFuture;
+    private long refreshRate;
+    private boolean propertiesInitialized;
 
-    public BoschIndegoHandler(Thing thing) {
+    public BoschIndegoHandler(Thing thing, HttpClient httpClient) {
         super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Indego handler");
+        BoschIndegoConfiguration config = getConfigAs(BoschIndegoConfiguration.class);
+        String username = config.username;
+        String password = config.password;
+
+        if (username == null || username.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error.missing-username");
+            return;
+        }
+        if (password == null || password.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error.missing-password");
+            return;
+        }
+
+        controller = new IndegoController(httpClient, username, password);
+        refreshRate = config.refresh;
+
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduleJob();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Indego handler");
+        cancelJob();
+    }
+
+    private void scheduleJob() {
+        ScheduledFuture<?> pollFuture = this.pollFuture;
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
```

Comment:
Normally the job should not be already scheduled when this method is called. 
You could just schedule it if the variable is null. 
Or if you really want to cancel the job to reschedule it, why not calling your method cancelJob?

Reply:
This is only for safety to prevent a situation where a job would be running wild without a reference. Previously the method (now renamed) was called from multiple places, now only from `initialize()`. In case it would be rescheduled from some other method in the future, IMHO is safer to keep this logic.

`cancelJob()` is intended for `dispose()` and will interrupt job is already running. `scheduleJob()` will not.

Reply:
This is not the most common pattern. 
Why would you like to reschedule the job in case it would be already scheduled? 

Some examples with different pattern:

https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java#L595

https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java#L242

Reply:
Done.